binding.cc

```diff
@@ -13,8 +13,20 @@
     return NULL; \
   }
 
-#define NAPI_MAKE_CALLBACK_AND_ALLOC(env, nil, ctx, cb, n, argv, res, nread) \
-  if (napi_make_callback(env, nil, ctx, cb, n, argv, &res) == napi_pending_exception) { \
+#define NAPI_MAKE_CALLBACK_AND_ALLOC(env, nil, ctx, cb, n, argv, nread) \
+  napi_value res; \
+  napi_status stat = napi_make_callback(env, nil, ctx, cb, n, argv, &res); \
+  if (stat == napi_ok) { \
+    bool is_buf; \
+    napi_is_buffer(env, res, &is_buf); \
+    if (is_buf) { \
```

Comment:
I added this to make the logic simpler on the JS (not requiring the EMPTY buffer) and also easier to understand. During a debugging session I was puzzling what is going on and this made it much easier to understand.

Reply:
Could you leave this as it was before? Lots of tuning and debugging has gone into the native layer and it works quite well atm

Reply:
removed in https://github.com/mafintosh/utp-native/pull/42/commits/54790942507088de2af7d2368fc6e32290d9f4f1